apiserver/test/e2e/cluster_server_e2e_test.go

```diff
@@ -516,38 +517,123 @@ func TestDeleteCluster(t *testing.T) {
 	}
 }
 
+func createClustersInNamespaces(t *testing.T, numberOfNamespaces int) []*End2EndTestingContext {
+	tCtxs := make([]*End2EndTestingContext, numberOfNamespaces)
+	var wg sync.WaitGroup
+	wg.Add(numberOfNamespaces)
+	for i := 0; i < numberOfNamespaces; i++ {
+		go func(i int) {
+			defer wg.Done()
+			tCtx, err := NewEnd2EndTestingContext(t)
+			require.NoError(t, err, "No error expected when creating testing context")
```

Comment:
```suggestion
			assert.NoError(t, err, "No error expected when creating testing context")
```
We can't use `require.NoError` in a goroutine created during the test because it uses `FailNow` under the hood. See: https://cs.opensource.google/go/go/+/refs/tags/go1.24.2:src/testing/testing.go;l=1011-1013